engine/orchestration/src/main/java/org/apache/cloudstack/engine/datacenter/entity/api/db/EngineHostVO.java

```diff
@@ -730,7 +732,9 @@ public boolean isDisabled() {
 
     @Override
     public boolean isInMaintenanceStates() {
-        return (getResourceState() == ResourceState.Maintenance || getResourceState() == ResourceState.ErrorInMaintenance || getResourceState() == ResourceState.PrepareForMaintenance);
+        ResourceState state = getResourceState();
+        return (state == ResourceState.Maintenance || state == ResourceState.ErrorInMaintenance ||
```

Comment:
This may further be simplified by using `Arrays().contains`

Reply:
Done